mdns.md

```diff
@@ -0,0 +1,362 @@
+mDNS; users will just be unable to discover any presentation displays.
+
+[Issue #37: Share data on mDNS reliability](issues/37)
+
+## Record caching
+
+One interesting aspect of mDNS is the ability for intervening layers of software
+between the controller and the presentation display (such as the underlying
+controller OS or the router firmware) to cache mDNS records and respond to
+queries, even if the original presentation display is unable to communicate
+directly with the controller.  If this is implemented with correct support for
+cache coherency, this may improve reliability by making mDNS more tolerant of
+transient network interruptions.
+
+# Latency of device discovery / device removal
+
+When a presentation display connects to a network, it must advertise itself by
+repeately multicasting its resource records with the "cache flush" bit set to 1
+([RFC 6762 Section 8.3](https://tools.ietf.org/html/rfc6762#section-8.3)).  This
+should give listeners the opporunity to discover newly available displays
+shortly after they are attached to a network.  However, to meet the requirements
+of the specification, a probing step must be completed first to break any naming
+ties, which will introduce a delay.
+
+When presentation display is disconnected from a network, it should similarly
+advertise a "goodbye" packet that updates its resource records with a TTL of
+zero ([RFC 6762 Section 10.1](https://tools.ietf.org/html/rfc6762#section-10.1)).
+If things "work as expected", listeners will delete these records from
+their cache on receipt and promptly show the presentation display as
+unavailable.
+
+If the presentation display is abruptly disconnected from the network and is not
+able to transmit a goodbye packet, listeners must wait for the expiration of
+their cached records to show the display as unavailable.  This depends on the
+TTL set with these records.
+
+# Network and power efficiency
+
+The network utilization of mDNS depends on several factors including:
+* The number of controllers and presentation displays connected to the network
+  at any one time.
+* The frequency that devices enter and leave the network.
+* TTL for mDNS answers and cache behavior on listeners.
+* The use of Known Answer suppression, which allows listeners to include
+  currently cached resource records with their queries.
+* Whether all answer records can fit into a single packet.
+
+Because mDNS behavior is specified, it should be possible to construct a model
+of network activity given assumptions about the factors above.
+
+[Issue #38: Obtain data on mDNS network and power efficiency](issues/38)
+
+# Ease of implementation / deployment
+
+## Platform support
+
+mDNS has been supported on Mac OS X since version 10.2 (released in 2002) under
+the brand name Rendezvous (now Bonjour), and iOS since its initial release
+in 2007.  It has been supported in Android since version 4.1 (Jelly Bean,
+released in 2012).
```

Comment:
Later version of Windows 10 also provide a system-wise mDNS service as well.
https://docs.microsoft.com/en-us/uwp/api/Windows.Networking.ServiceDiscovery.Dnssd